src/bokeh/models/glyphs.py

```diff
@@ -1208,7 +1208,13 @@ def __init__(self, *args, **kwargs) -> None:
     """)
 
 class Rect(XYGlyph, LineGlyph, FillGlyph, HatchGlyph):
-    ''' Render rectangles.
+    ''' Render rectangles, characterised by center position (x and y), width,
+    height, and angle of rotation.
+
+    .. warning::
```

Comment:
@mattpap I don't personally see the connection to #12421 This is not a case of experimental feature, it's simply an inherent mathematic limitation that a specification for " center rect" shapes is not well-defined on a log scale (could mean multiple things, none especially good or obviously "correct"). That's not changing in the future.

I'm neither for nor against a validation warning, I guess it would be useful for folks trying to use rects on log scales, but I also don't think many people at all try to do that. If @ianthomas23 want's to add one that's fine with me, but I would not hold up this PR for lack of one. (However I am surprised to hear you advocate for one since I had the impression you disfavored the validation mechanism, generally).

Reply:
The connection is the conclusion that people don't pay attention to documentation. I think this is actually a good candidate for having a runtime warning. One thing is that there are setups that make sense (log axes and screen units make perfect sense) and some that make less or no sense, and one can easily go from what makes sense to what doesn't. Another thing is that `Rect` is unlikely to be used directly, but `Figure.rect()` is more likely to be used. So, either this warning should be appended to `rect()`'s docstring as well, or we should have a runtime warning.

> I am surprised to hear you advocate for one since I had the impression you disfavored the validation mechanism, generally).

Not really, I just don't like too much verboseness by default. One thing, however, I'm definitely for is consistency in behavior, and I think that having a runtime warning here is consistent with previous decisions.

Reply:
> people don't pay attention to documentation. 

This is true in some sense, but also an exaggeration. But regardless one of the big benefits of thorough documentation is for devs: having an answer to link directly to when questions do come up

+1 to including info in `rect` as well as `Rect` 

I'm still +0 on requiring a validation check mostly because the recent revelation about `select` has me cautious about the cost of model graph traversals (which a validation here would probably require)